proteus/SpatialTools.py

```diff
@@ -0,0 +1,1469 @@
+"""
```

Comment:
I think we should probably rename this mprans/SpatialTools.py or something like that. There is a lot of functionality in here, but much of it is specific to the two-phase modeling and rigid  body modeling.